### PR TITLE
Parse GitHub from website links

### LINKED ISSUE #37: Parse websites and social links
e.g. see https://orcid.org/0000-0002-5292-4083

### PR DESCRIPTION
**Pull request by cthoyt**
Closes #37 

Anyone want to flesh this out with additional logic for other types? e.g. twitter?

### CODE REVIEW

**Comment by jvfe** [comment]
Awesome! I believe this logic could be adapted and generalized for other types (i.e. twitter, gitlab, etc). So thanks a lot!

File: src/pyorcidator/helper.py [hunk 1]
```diff
@@ -18,15 +19,25 @@
...
+def get_external_ids(data) -> Mapping[str, str]:
+    """Get external identifiers that can be mapped to Wikidata properties."""
+    rv = {}
+    for d in data["person"]["external-identifiers"]["external-identifier"]:
+        rv[d["external-id-type"]] = d["external-id-value"]
+    for d in data["person"]["researcher-urls"].get("researcher-url", []):
+        url_name = d["url-name"].lower().replace(" ", "")
+        url = d["url"]["value"].rstrip("/")
+        if url_name == "github" and "github.com" in url:
+            github = url[len("https://github.com/")]
```
Comment (jvfe) [inline src/pyorcidator/helper.py]: I've noticed there's a small bug here. You probably want `github = url[len("https://github.com/"):]`. Notice the colon

**Comment by cthoyt** [comment]
@jvfe feel free to push to either this branch or we can finish up here and you can continue in your own - whatever's more comfortable

File: tests/test_helper.py [hunk 2]
```diff
@@ -57,3 +58,11 @@ def test_get_loop_id(sample_orcid_data):
...
 def test_render_orcid_runs():
     render_orcid_qs("0000-0003-4423-4370")
+
+
+def test_get_github():
+    """Test getting a github link."""
+    orcid = "0000-0002-0791-1347"  #  Selja Seppälä
+    data = get_orcid_data(orcid)
+    ids = get_external_ids(data)
+    assert ids["github"] == "seljaseppala"
```
Comment (jvfe) [inline tests/test_helper.py:68]: at the moment this returns just 's' when running `pyorcidator import --orcid 0000-0002-0791-1347`

**Review by jvfe — CHANGES_REQUESTED** [review]

**Comment by jvfe** [comment]
> @jvfe feel free to push to either this branch or we can finish up here and you can continue in your own - whatever's more comfortable

For sure! I'll take a look as soon as I'm able.

File: src/pyorcidator/helper.py (hunk 1 above)
Comment (cthoyt) [reply src/pyorcidator/helper.py]: Indeed, I think I fixed this in c9e7f9c75180cd50350e099d613d778c9f74a82f

File: tests/test_helper.py (hunk 2 above)
Comment (cthoyt) [reply tests/test_helper.py:68]: yes also related to the bug you mentioned, which I believe should be fixed now. Are you on the most recent commit? Note that the test is passing on CI now.

File: src/pyorcidator/helper.py (hunk 1 above)
Comment (jvfe) [reply src/pyorcidator/helper.py]: Yep, thanks!

**Review by jvfe — APPROVED** [review]

**Review by jvfe — APPROVED** [review]
Great way of generalizing this! I think this can be merged as is. It's already a big improvement and adding further external links will be very simple.

File: src/pyorcidator/helper.py [hunk 3]
```diff
@@ -18,15 +19,30 @@
...
+    ("github", "https://github.com/"),
+    ("twitter", "https://twitter.com/"),
+    ("scopus", "https://www.scopus.com/authid/detail.uri?authorId=")
+    # TODO linkedin, figshare, researchgate, publons, semion, semantic scholar, google scholar, etc.
+]
+
+
+def get_external_ids(data) -> Mapping[str, str]:
+    """Get external identifiers that can be mapped to Wikidata properties."""
+    rv = {}
```
Comment (lubianat) [inline src/pyorcidator/helper.py:36]: what does rv mean?

Comment (cthoyt) [reply src/pyorcidator/helper.py:36]: return value

Comment (cthoyt) [reply src/pyorcidator/helper.py:36]: This is a questionable habit I picked up when reading through variuos Pocoo projects like click and flask

**Comment by cthoyt** [comment]
@lubianat can you change the project settings to enforce squash/merge and also auto-delete branches after merge?

**Comment by lubianat** [comment]
I can try, not sure how to do it though
Tiago

**Comment by cthoyt** [comment]
Go to https://github.com/lubianat/pyorcidator/settings, scroll about 2/3 the way down, then you will see the following settings (below is the Bioregistry repo)

Screen Shot 2022-10-12 at 13 47 54

File: src/pyorcidator/helper.py (hunk 3 above)
Comment (lubianat) [reply src/pyorcidator/helper.py:36]: Oh, I see, thanks! Yeah, I'd vote for us to avoid these contractions and use plain English

Comment (cthoyt) [reply src/pyorcidator/helper.py:36]: Agreed, thanks for being vigilant! I will do better :)

**Pull request merged**